internal/util/observability/funccall.go

```diff
@@ -56,11 +58,16 @@ func FuncCall(ctx context.Context) func() {
 		fc.region = trace.StartRegion(ctx, funcName)
 	}
 
-	return fc.leave
+	stopCtx, stopCancel := context.WithCancel(context.Background())
```

Comment:
We create new context instead of using ctx passed in this function on purpose right? Why is that?

Reply:
Yes, it was written on purpose. This context is not related to the initial context. Before this PR,`FuncCall` returned a function that should be called at the end of the caller function (so, it was deferred). One of the ideas here was to use context with cancelation instead of that deferred function, so this context is only related to cancelation. 

If we pass the original context `ctx` instead of `context.Background()`, and `ctx` is canceled, this `stopCtx` will be canceled before exiting from the caller function, my idea was to avoid it and only have `stopCtx` canceled when `stopCancel` is called.

Reply:
That causes values inside original `ctx` to be lost